src/java.base/share/classes/java/net/HttpURLConnection.java

```diff
@@ -606,11 +606,20 @@ public long getHeaderFieldDate(String name, long defaultValue) {
      * @return a {@code SocketPermission} object representing the
      *         permission necessary to connect to the destination
      *         host and port.
+     *
+     * @deprecated
+     * Permissions can no longer be used for controlling access to resources
+     * as the Security Manager is no longer supported. This method may be
+     * changed in the future to unconditionally throw
+     * {@link UnsupportedOperationException}, or return
+     * {@link java.security.AllPermission}.
      */
+    @Deprecated(since = "25")
```

Comment:
can this (and all other getPermission methods) be "forRemoval=true"?

Reply:
Possibly. @AlanBateman may have some thoughts on this.

Reply:
I think we should at least consider deprecating URLConnection.getPermission and HttpURLConnection.getPermission for removal, only because I don't immediately see the advantage of doing it in two steps for these two APIs.

Reply:
Thanks - I will do that then

Reply:
Done